### PR TITLE
EOS-27248: support for Mock Upgrade Miniprovisioner for HA

### PR DESCRIPTION
**Pull request by tanujashinde0405**
Signed-off-by: Tanuja Shinde <tanuja.shinde@seagate.com>

# Problem Statement
- EOS-27248: added support for Mock Upgrade Miniprovisioner for HA in ha_setup 

# Design
-  Ticket ref: https://jts.seagate.com/browse/EOS-27248 

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: N
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]: N
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [x] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: Y

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by ArchanaLimaye** [comment]
1. Please update the checklist and other sections in  PR description
2. Please update testing details in the PR

File: ha/k8s_setup/ha_setup.py [hunk 1]
```diff
@@ -306,6 +306,24 @@ def process(self):
         """
         sys.stdout.write('HA initialization is done.\n')
 
+class UpgradeCmd(Cmd):
```
Comment (Madhura-08) [inline ha/k8s_setup/ha_setup.py:309]: Can you please add "upgrade" in the usage() function?

Comment (tanujashinde0405) [reply ha/k8s_setup/ha_setup.py:309]: done

**Review by shaileshsvc — APPROVED** [review]
Approved changes for upgrade stub.

**Review by ArchanaLimaye — APPROVED** [review]

**Review by Madhura-08 — APPROVED** [review]

**Pull request merged**